src/core/statistics/useStatistics.ts

```diff
@@ -30,39 +38,98 @@ export type TrackParameters = {
 
 export type EventAction = "send";
 
-export function useStatistics() {
+export const useCollectPageStatistics = () => {
```

Comment:
I would appreciate some of the commentary from this pull request about the purpose of each of these hooks.  This might include information about why `useCollectPageStatistics()` and `usePageStatistics()` is separated.

Reply:
I tried here:

The reason these concerns are split into separate hooks is to keep usage flexible:

usePageStatistics needs access to Mapp config values in order to inject the tiLoader script, so it’s can only used in contexts where those values are available.
useCollectPageStatistics is intentionally kept decoupled from the config, making it usable in components that only need to collect data without triggering script injection.
useTrackStatistics (Now useEventStatistics) relies on a separate tiLoader integration that comes from the Drupal layer, and does not depend on the local Mapp config.